File: Abeyonix_Frontend/src/types/companySettings.ts
export interface CompanySettingsResponse {
  id: number;

  company_name?: string;
  company_email?: string;
  company_phone?: string;

  address_line1?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  country?: string;

  gst_number?: string;
  website?: string;

  logo_url?: string;
}

export interface CompanySettingsPayload {
  company_name: string;
  company_email?: string;
  company_phone?: string;

  address_line1?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  country?: string;

  gst_number?: string;
  website?: string;

  logo?: File | null;
}

File: Abeyonix_Backend/app/schemas/company_settings.py
from pydantic import BaseModel, EmailStr
from typing import Optional


class CompanySettingsBase(BaseModel):
    company_name: Optional[str] = None
    company_email: Optional[EmailStr] = None
    company_phone: Optional[str] = None

    address_line1: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    postal_code: Optional[str] = None
    country: Optional[str] = None

    gst_number: Optional[str] = None
    website: Optional[str] = None
    logo_url: Optional[str] = None


class CompanySettingsCreate(CompanySettingsBase):
    pass


class CompanySettingsUpdate(CompanySettingsBase):
    pass


class CompanySettingsResponse(CompanySettingsBase):
    id: int

    class Config:
        from_attributes = True